yagb-core/src/emulator/infrared.ts

```diff
@@ -0,0 +1,26 @@
+import {Bus, ReadHandler, WriteHandler} from "./bus";
+import {Savestate} from "./savestate";
+import {cgbRegisters} from "./cgb-registers";
+import {Mode} from "./mode";
+
+export class Infrared {
+    constructor(private mode: Mode) {}
+
+    install(bus: Bus): void {
+        if(Mode.cgb === this.mode) {
```

Comment:
I'd prefer an early return here, although that's arguable just a matter of taste 😏